### PR TITLE
Widen glob matching to include suffix-style tests

### LINKED ISSUE #1: Glob patterns do not catch tests in the form of "filename.spec.extension" or "filename.test.extension"
Hello again,

It looks like the issue with this module is that it's not catching tests in the following form(s)

```
src\
├── index.js
├── index.test.js // not caught
├── index.spec.js // not caught
```
It looks like the globbing requires the test files to be in a folder named `test`, `spec`, or `__tests__`, which I can tell you is not always the case.

Please consider extending the glob pattern.

Thanks

### PR DESCRIPTION
**Pull request by dsifford**
Closes #1 

Had to add an additional line of globbing to prevent `src/foo/latest.js`, `src/foo/fastest.js`, etc from matching. Let me know if there's a clever way to combine if you know of one and would like me to fix.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -15,7 +15,8 @@ function detectRepoTestFiles(dir, options) {
 
     return globby([
         '**/{test,spec}?(s)/**/*',            // Conventional test dirs (including deep)
-        '{test,spec}?(s).*',                  // Conventional root test files
+        '**/*[.,-]{test,spec}?(s).*',         // Suffix-style test files (foo-test.js, foo.test.js)
```
Comment (satazor) [inline index.js]: I think the `,` is unnecessary no?

Comment (dsifford) [reply index.js]: Ah, yep you're right... Looks like I hybridized the `{}` and `[]` patterns 😝 

Adjusting now....

File: index.js [hunk 2]
```diff
@@ -15,7 +15,8 @@ function detectRepoTestFiles(dir, options) {
 
     return globby([
         '**/{test,spec}?(s)/**/*',            // Conventional test dirs (including deep)
-        '{test,spec}?(s).*',                  // Conventional root test files
+        '**/*[.-]{test,spec}?(s).*',          // Suffix-style test files (foo-test.js, foo.test.js)
```
Comment (satazor) [inline index.js]: Perhaps we should also support `_` here? Sorry for only saying this now.

Comment (dsifford) [reply index.js]: No problem. I'll add that now.

**Pull request merged**